first_diff.py

```diff
@@ -1,255 +1,50 @@
-            continue
-        if addr[0] != map2[sym][0]:
-            if min_ram is None or addr[0] < min_ram:
-                min_ram = addr[0]
-                found = (sym, addr[1], addr[2])
-    if min_ram is None:
-        return False
-    else:
-        print(
-            f"Map appears to have shifted just before {found[0]} ({found[1]}) -- in {found[2]}?"
-        )
-        if found[2] is not None and found[2] not in map2:
-            print(
-                f"(Base map file {basemap} out of date due to new or renamed symbols, so result may be imprecise.)"
-            )
-        return True
-
-
-def hexbytes(bs):
-    return ":".join("{:02X}".format(c) for c in bs)
-
-
-found_instr_diff = []
-map_search_diff = []
-diffs = 0
-shift_cap = 1000
-for i in range(24, len(mybin), 4):
-    # (mybin[i:i+4] != basebin[i:i+4], but that's slightly slower in CPython...)
-    if diffs <= shift_cap and (
-        mybin[i] != basebin[i]
-        or mybin[i + 1] != basebin[i + 1]
-        or mybin[i + 2] != basebin[i + 2]
-        or mybin[i + 3] != basebin[i + 3]
-    ):
-        if diffs == 0:
-            print(f"First difference at ROM addr 0x{i:X}, {search_rom_address(i)}")
-            print(
-                f"Bytes: {hexbytes(mybin[i : i + 4])} vs {hexbytes(basebin[i : i + 4])}"
-            )
-        diffs += 1
-    if (
-        len(found_instr_diff) < diff_count
-        and mybin[i] >> 2 != basebin[i] >> 2
-        and not search_rom_address(i) in map_search_diff
-    ):
-        found_instr_diff.append(i)
-        map_search_diff.append(search_rom_address(i))
-
-if diffs == 0:
-    print("No differences but ROMs differ?")
-    exit()
-
-if len(found_instr_diff) > 0:
-    for i in found_instr_diff:
-        print(f"Instruction difference at ROM addr 0x{i:X}, {search_rom_address(i)}")
-        print(
-            f"Bytes: {hexbytes(mybin[i : i + 4])} vs {hexbytes(basebin[i : i + 4])}"
-        )
-print()
-
-definite_shift = diffs > shift_cap
-if definite_shift:
-    print(f"Over {shift_cap} differing words, must be a shifted ROM.")
-else:
-    print(f"{diffs} differing word(s).")
-
-if diffs > 100:
-    if not os.path.isfile(basemap):
-        print(
-            f"To find ROM shifts, copy a clean .map file to {basemap} and rerun this script."
-        )
-    elif not map_diff():
-        print(f"No ROM shift{' (!?)' if definite_shift else ''}")
-
-if args.diff_args:
-    if len(found_instr_diff) < 1:
-        print(f"No instruction difference to run diff.py on")
-        exit()
-
-    diff_sym = search_rom_address(found_instr_diff[0]).split()[0]
-    if args.diff_args == "prompt":
-        diff_args = input("Call diff.py with which arguments? ") or "--"
-    else:
-        diff_args = args.diff_args
-    if diff_args[0] != "-":
-        diff_args = "-" + diff_args
-    check_call(
-        [
-            "python3",
-            "diff.py",
-            diff_args,
-            diff_sym,
-        ]
-    )
+import mapfile_parser
```

Comment:
one request:
```suggestion
try:
    import mapfile_parser
except ImportError:
    print("Missing dependency mapfile_parser, install it with `python3 -m pip install mapfile-parser`")
    exit(1)
```
(written in github, didn't test, there may be an obvious mistake)